### PR TITLE
update to 5.rc1

### PR DESCRIPTION
**Pull request by soumyaDghosh**

### CODE REVIEW

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -27,59 +21,22 @@ apps:
...
-      - gettext
-      - libxml2-dev
-    override-pull: |
-      set -eux
-      snapcraftctl pull
-      sed -i 's#Werror=missing-include-dirs#Wmissing-include-dirs#g' meson.build
+    meson-parameters: [ --prefix=/usr, -Dgir=false, -Dvapi=false, -Dgtk_doc=false]
     prime:
-      - usr/lib/*/libgtksourceview*
+      - -usr/lib/*/pkgconfig
```
Comment (kenvandine) [inline snap/snapcraft.yaml:37]: We shouldn't stage pkgconfig files

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -27,59 +21,22 @@ apps:
     common-id: org.gnome.Fractal
     desktop: usr/share/applications/org.gnome.Fractal.desktop
     plugs:
-      - home
-      - removable-media
       - network
+      - network-bind
```
Comment (kenvandine) [inline snap/snapcraft.yaml]: Are you sure we need network-bind?

**Review by kenvandine — COMMENTED** [review]
This builds fine and works better than 4.x version, but it is crashing when clicking links

File: snap/snapcraft.yaml (hunk 1 above)
Comment (soumyaDghosh) [reply snap/snapcraft.yaml:37]: Are staging it? I think it's 

`- -usr/lib/*/pkgconfig`

Isn't it meant to not stage it?

Comment (soumyaDghosh) [reply snap/snapcraft.yaml:37]: Uhh.. Got it.. You mean that instead `prime` I shoild do `stage` right?

Comment (kenvandine) [reply snap/snapcraft.yaml:37]: oh, sorry... miss-read that.  It's fine :)

File: snap/snapcraft.yaml (hunk 2 above)
Comment (soumyaDghosh) [reply snap/snapcraft.yaml]: This is tbh not sure of. But, there was one app I tried, it took more time to use the internet without `network-bind`. So, I simply added it here also. But, as you said, I should look into it.

**Comment by soumyaDghosh** [comment]
> This builds fine and works better than 4.x version, but it is crashing when clicking links

Okay, so, I got the problem. It's not this snap specific. When you run a snap from the terminal and click on a link it'll crash. Try to run from the desktop file and click an url. Everything will be fine there. But, I am not sure why is this happening.

**Comment by kenvandine** [comment]
5.rc1 has been tagged, can you please try updating to that tag and we'll give it another test.

**Comment by soumyaDghosh** [comment]
Probably this time, I need to build `pipewire` from source. Getting this everytime I try to build

```
:: The following warnings were emitted during compilation:
::
:: warning: src/type-info.c:61:58: error: ‘spa_type_video_flags’ undeclared here (not in a function); did you mean ‘spa_type_audio_flags’?
:: warning:    61 | const struct spa_type_info* libspa_rs_type_video_flags = spa_type_video_flags;
:: warning:       |                                                          ^~~~~~~~~~~~~~~~~~~~
:: warning:       |                                                          spa_type_audio_flags
:: warning: src/type-info.c:62:67: error: ‘spa_type_video_interlace_mode’ undeclared here (not in a function); did you mean ‘spa_video_interlace_mode’?
:: warning:    62 | const struct spa_type_info* libspa_rs_type_video_interlace_mode = spa_type_video_interlace_mode;
:: warning:       |                                                                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
:: warning:       |                                                                   spa_video_interlace_mode
::
:: error: failed to run custom build command for `libspa-sys v0.7.2`
::
:: Caused by:
::   process didn't exit successfully: `/build/snapcraft-fractal-671354de5a2654c1f1988e4e8c3e2d43/parts/fractal/build/src/release/build/libspa-sys-33b60a6459fa6cbe/build-script-build` (exit status: 1)
```

https://launchpadlibrarian.net/697585084/buildlog_snap_ubuntu_jammy_amd64_snapcraft-fractal-671354de5a2654c1f1988e4e8c3e2d43_BUILDING.txt.gz

Should I try to build pipewire from source, or try to add it in `gnome-sdk` itself?

**Comment by kenvandine** [comment]
You can copy the pipewire part from my WIP snapshot snap

https://github.com/kenvandine/snapshot/blob/main/snap/snapcraft.yaml#L31

**Comment by soumyaDghosh** [comment]
> You can copy the pipewire part from my WIP snapshot snap
> 
> https://github.com/kenvandine/snapshot/blob/main/snap/snapcraft.yaml#L31

I have it too, made it for `gnome-sdk` itself. BTW, I also had snapshot. But, it needs access to pipewire-lock file. Should I share that manifest?

**Comment by soumyaDghosh** [comment]
Done. But, got this. 

```
2023-11-13T20:09:22.670705Z ERROR fractal::secret: Failed to move database: Permission denied (os error 13)
```
App works fine though, (some chats weren't decrypted though)
Probably we need to give the option of `password-manager-service`. 

EDIT: kindly check the denials of `snappy-debug`. Probably looking for cpus max core or thread or something like that. Can it hamper the performance of the app?

File: snap/snapcraft.yaml (hunk 2 above)
Comment (kenvandine) [reply snap/snapcraft.yaml]: Please remove network-bind, it shouldn't be needed

File: snap/snapcraft.yaml [hunk 3]
```diff
@@ -99,54 +57,90 @@ parts:
...
-      - gstreamer1.0-plugins-base
-      - gstreamer1.0-plugins-good
-      - gstreamer1.0-plugins-ugly
-      - libges-1.0-0
-      - libgstreamer1.0-0
-      - libgstreamer-plugins-bad1.0-0
-      - libgstreamer-plugins-base1.0-0
-      - libgstreamer-plugins-good1.0-0
-      - libhandy-0.0-0
+      - libpipewire-0.3-dev
```
Comment (sergiusens) [inline snap/snapcraft.yaml]: you can remove this, you are already building it

Comment (soumyaDghosh) [reply snap/snapcraft.yaml]: Ahh.. sorry, missed it

File: snap/snapcraft.yaml (hunk 2 above)
Comment (soumyaDghosh) [reply snap/snapcraft.yaml]: Done

**Review by kenvandine — APPROVED** [review]
Thanks!

**Comment by sergiusens** [comment]
Might not be due to this PR, but it seems we are missing gst plugins

```
Your GStreamer installation is missing a plug-in.
../gst/playback/gsturidecodebin.c(1027): no_more_pads_full (): /GstPlayBin:playbin/GstURIDecodeBin:uridecodebin0:
no suitable plugins found:
../gst/playback/gstdecodebin2.c(4701): gst_decode_bin_expose (): /GstPlayBin:playbin/GstURIDecodeBin:uridecodebin0/GstDecodeBin:decodebin0:
no suitable plugins found:
Missing decoder: MPEG-4 AAC (audio/mpeg, mpegversion=(int)4, framed=(boolean)true, stream-format=(string)raw, level=(string)2, base-profile=(string)lc, profile=(string)lc, codec_data=(buffer)1208, rate=(int)44100, channels=(int)1)
Missing decoder: H.264 (Main Profile) (video/x-h264, stream-format=(string)avc, alignment=(string)au, level=(string)3.2, profile=(string)main, codec_data=(buffer)014d0020ffe1001b674d4020eca05a05fd3505060540000003004000000f03c60c658001000468ef8fc8, width=(int)720, height=(int)1520, framerate=(fraction)30/1, pixel-aspect-ratio=(fraction)1/1)
```

**Comment by soumyaDghosh** [comment]
> Might not be due to this PR, but it seems we are missing gst plugins
> 
> ```
> Your GStreamer installation is missing a plug-in.
> ../gst/playback/gsturidecodebin.c(1027): no_more_pads_full (): /GstPlayBin:playbin/GstURIDecodeBin:uridecodebin0:
> no suitable plugins found:
> ../gst/playback/gstdecodebin2.c(4701): gst_decode_bin_expose (): /GstPlayBin:playbin/GstURIDecodeBin:uridecodebin0/GstDecodeBin:decodebin0:
> no suitable plugins found:
> Missing decoder: MPEG-4 AAC (audio/mpeg, mpegversion=(int)4, framed=(boolean)true, stream-format=(string)raw, level=(string)2, base-profile=(string)lc, profile=(string)lc, codec_data=(buffer)1208, rate=(int)44100, channels=(int)1)
> Missing decoder: H.264 (Main Profile) (video/x-h264, stream-format=(string)avc, alignment=(string)au, level=(string)3.2, profile=(string)main, codec_data=(buffer)014d0020ffe1001b674d4020eca05a05fd3505060540000003004000000f03c60c658001000468ef8fc8, width=(int)720, height=(int)1520, framerate=(fraction)30/1, pixel-aspect-ratio=(fraction)1/1)
> ```

This is the same issue, that I have never been able to fix. May be Identity snap manifest might help

**Pull request merged**